internal/daemon/controller/handlers/credentials/credential_service.go

```diff
@@ -845,15 +845,15 @@ func validateUpdateRequest(req *pbs.UpdateCredentialRequest) error {
 		case credential.JsonSubtype:
 			object := req.GetItem().GetJsonAttributes().GetObject()
 			if object != nil {
-				objectBytes, err := json.Marshal(object.AsMap())
+				_, err := json.Marshal(object.AsMap())
 				if err != nil {
 					badFields[objectField] = "Unable to parse given json value"
 				}
-				if handlers.MaskContains(req.GetUpdateMask().GetPaths(), objectField) && len(objectBytes) <= 0 {
+				if !handlers.MaskContainsSubString(req.GetUpdateMask().GetPaths(), objectField) || len(object.AsMap()) <= 0 {
```

Comment:
I wonder if it would make the code easier to maintain and provide a more clear error message to the user if we break this into two checks. The first checks if the mask was provided and the second checks if the len is == 0.

Also nit `len(object.AsMap()) == 0` instead of the `<=`

Reply:
I agree, this will make it easier to read the conditional statement.